File: Aplikacja/Mobile/OtoAuto/app/src/main/java/org/masters_thesis/otoauto/model/OfferCardComponentModel.kt
package org.masters_thesis.otoauto.model

data class OfferCardComponentModel(
    var offerId: Int,
    var engineCapacity: String,
    var fuelType: String,
    var mileage: String,
    var mileageUnit: String,
    var offerCurrency: String,
    var offerMainImage: OfferImage,
    var offerPrice: String,
    var offerTitle: String,
    var yearOfProduction: String
)

File: Aplikacja/VueApp/oto-auto/src/store/model/offer-card-component.model.ts
import { OfferImage } from "./offer-image.model";
import { VehicleDetailsRow } from "./vehicle-details-row.model";

export interface OfferCardComponentModel {
    offerId: number;
    offerMainImage: OfferImage,
    offerTitle: string,
    yearOfProduction: string,
    mileage: string,
    mileageUnit: string,
    fuelType: string,
    engineCapacity: string,
    offerPrice: string,
    offerCurrency: string
}

export interface OfferActivityComponentModel {
    offerId: number;
    offerImages: OfferImage[],
    offerTitle: string,
    offerDescription: string
    yearOfProduction: string,
    mileage: string,
    fuelType: string,
    engineCapacity: string,
    offerPrice: string,
    offerCurrency: string
    equipments: string[]
    vehicleAttributes: VehicleDetailsRow[]
}

export default interface OfferCardComponentStateModel {
    offerCardsComponent: OfferCardComponentModel[],
    offer: OfferActivityComponentModel
    offerImages: OfferImage[]
    redisNotResponding: boolean
}